visualdl/backend/storage/storage.proto

```diff
@@ -37,11 +37,43 @@ message Entry {
     repeated float fs = 9;
     repeated double ds = 10;
     repeated int32 i32s = 11;
-    repeated bool bs = 12;
+    repeated string ss = 12;
+    repeated bool bs = 13;
 }
 
+/*
+The Record proto is designed to represent any data structure for any component, for
+example, to store a record of Scalar component
+
+Record {
+  // training error is 0.1, testing error is 0.2
+  data = [0.1, 0.2],
+  timestamp = xxxx,
+  // step id
+  id = xxxx
+}
+
+to store a record of Image component
+
+Record {
+  // RBG pixel weights of a image
+  data = [[0.1, 0.2, ...], [...], [...]],
+  timestamp = xxxx,
+  // image shape
+  meta = [100, 200]
+}
+
+for other complex structure which have more fields than `timestamp`, `id` and meta, the additional fields
+can store in the `data` field, for it can store a list of values in different basic data types.
+
+A component handlers in logic layer should know how to write or load a record for the corresponding component.
+*/
 message Record {
-    Entry data = 1;
+    // one record might have multiple fields, one specific component should know how
+    // to parse the records.
+    repeated Entry data = 1;
+    // NOTE the timestamp, id, dtype might be useless for that all the meta info can
+    // be stored in `data` field.
     int64 timestamp = 2;
     // store the count of writing operations to the tablet.
```

Comment:
remove this comment

id should be set.

Reply:
rename num_records